### PR TITLE
Feat/text highlight plugin

### PR DESCRIPTION
**Pull request by talevy17**

### CODE REVIEW

File: packages/common/web/src/Utils/textUtils.jsx [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export const isJSON = text =>
```
Comment (lxgreen) [inline packages/common/web/src/Utils/textUtils.jsx]: i'd like to discuss this regex

Comment (talevy17) [reply packages/common/web/src/Utils/textUtils.jsx]: I'll be in office on Thursday

File: packages/plugin-text-color/web/src/toolbar/BaseTextColor.jsx [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+  helpers: PropTypes.object,
+  keyName: PropTypes.string,
+  anchorTarget: PropTypes.string,
+  relValue: PropTypes.string,
+  t: PropTypes.func,
+  tabIndex: PropTypes.number,
+  uiSettings: PropTypes.object,
+  config: PropTypes.object,
+  setKeepOpen: PropTypes.func,
+  decorator: PropTypes.object.isRequired,
```
Comment (lxgreen) [inline packages/plugin-text-color/web/src/toolbar/BaseTextColor.jsx]: `decorator` term is super-overused in `rich-content`, please find a better + meaningful name to this object

File: packages/plugin-text-color/web/src/text-decorations-utils.jsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  }
+  return type === 'FG' ? style : '';
+};
+
+export const getColor = style => {
+  const splitted = splitStyle(style);
+  return splitted.length === 2 ? splitted[1] : splitted[0];
+};
+
+export const textColorPredicateWrapper = styleSelectionPredicate => {
```
[the remark this answers is not in the document]
Comment (talevy17) [reply packages/plugin-text-color/web/src/text-decorations-utils.jsx]: I thought the naming would come as an issue, couldn't think of anything else.
Any suggestions?

File: packages/plugin-text-color/web/src/toolbar/TextHighlightIcon.jsx [hunk 4]
```diff
@@ -3,10 +3,20 @@ import React from 'react';
 
 const TextHighlightIcon = props => (
   <svg width="19px" height="19px" viewBox="0 0 19 19" {...props}>
-    <g stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
+    <title>Fill Color</title>
+    <g id="Fill-Color" stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
```
Comment (lxgreen) [inline packages/plugin-text-color/web/src/toolbar/TextHighlightIcon.jsx]: please try to remove all the `id` attributes since no one references them

**Pull request merged**